ddtrace/contrib/dbapi/__init__.py

```diff
@@ -164,7 +165,7 @@ def fetchmany(self, *args, **kwargs):
         )
 
 
-def _get_config(new_cfg):
+def _override_dbapi2_config(new_cfg):
```

Comment:
An alternative interim change, before the dbapi2 config is removed for good, could be
~~~ python
class OverrideAttrDict(wrapt.ObjectProxy):
    __slots__ = ("override", "base")
    sentinel = object()

    def __init__(self, override, base):
        super(OverrideAttrDict, self).__init__(override)
        self.override = override
        self.base = base

    def __getattr__(self, name):
        value = self.override.get(name, self.sentinel)
        return getattr(self.base, name) if value == self.sentinel else value

    def __getitem__(self, name):
        value = self.override.get(name, self.sentinel)
        return self.base.__getitem__(name) if value == self.sentinel else value


def _override_dbapi2_config(new_cfg):
    # Need to backwards support the dbapi2 config entry
    # but give precedence to the given config.

    return OverrideAttrDict(new_cfg, config.dbapi2)
~~~
Profiling data show that the overhead goes down to 0.1%

Reply:
Thanks for the tip @P403n1x87! I've added it in.

Reply:
@brettlangdon with @majorgreys' latest change, the total overhead of the new `_override_dbapi2_config` is around 0.3% (on average, per-request); with the old `_get_config`, the total overhead is around 6.3%.